datastream-server/src/main/java/com/linkedin/datastream/server/EventProducer.java

```diff
@@ -58,6 +58,8 @@ public class EventProducer implements DatastreamEventProducer {
 
   static final String EVENTS_LATENCY_MS_STRING = "eventsLatencyMs";
   static final String EVENTS_SEND_LATENCY_MS_STRING = "eventsSendLatencyMs";
+  static final String THROUGHPUT_VIOLATING_EVENTS_LATENCY_MS_STRING = "throughputViolatingEventsLatencyMs";
+  static final String THROUGHPUT_VIOLATING_EVENTS_SEND_LATENCY_MS_STRING = "throughputViolatingEventsSendLatencyMs";
```

Comment:
What about the other metrics from lines 67-70? Does the reporting need to be split for that as well?

```private static final String TOTAL_EVENTS_PRODUCED = "totalEventsProduced";
  private static final String EVENTS_PRODUCED_WITHIN_SLA = "eventsProducedWithinSla";
  private static final String EVENTS_PRODUCED_WITHIN_ALTERNATE_SLA = "eventsProducedWithinAlternateSla";

Reply:
For these latency violations, we can skip this metric EVENTS_PRODUCED_WITHIN_SLA, but I updated the code to emit EVENTS_PRODUCED_WITHIN_ALTERNATE_SLA metric as those would be applicable.